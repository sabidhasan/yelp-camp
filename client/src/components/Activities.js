import React from 'react'

const Activities = (props) => {
  if (!(props.activitiesList)) return null;

  return (
    <ul className='Activities'>
      {props.activitiesList.map((activity, idx) => (
        <li className='Activities__activity flex-center' key={idx}>
          <span className='Activities__logo'>{activity.logo}</span>
          <span className='Activities__name bold'>{activity.name}</span>
        </li>
      ))}
    </ul>
  )
}

export default Activities
