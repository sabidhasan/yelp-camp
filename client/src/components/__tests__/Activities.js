import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Activities from '../Activities'

configure({ adapter: new Adapter() });

describe('<Activities />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      activitiesList: [{logo: 's', name: 'aa'}, {logo: 's', name: 'aa'}],
    }
    component = shallow(<Activities { ...props } />)
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });


  it('Number of activities shown equal', () => {
    expect(component.find('.Activities__activity')).toHaveLength(2)
  })

  it('If no activites, return nothing', () => {
    component.setProps({activitiesList: null});
    expect(component.html()).toBe(null);
  })

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
})
