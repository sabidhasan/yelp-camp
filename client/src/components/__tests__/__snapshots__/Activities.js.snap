// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activities /> LoginForm component snapshot matches what is expected 1`] = `
<ul
  className="Activities"
  role="table"
>
  <li
    className="Activities__activity flex-center"
    key="0"
    role="cell"
  >
    <span
      className="Activities__logo"
    >
      s
    </span>
    <span
      className="Activities__name bold"
    >
      aa
    </span>
  </li>
  <li
    className="Activities__activity flex-center"
    key="1"
    role="cell"
  >
    <span
      className="Activities__logo"
    >
      s
    </span>
    <span
      className="Activities__name bold"
    >
      aa
    </span>
  </li>
</ul>
`;
